### DISCUSSION TITLE
Load Enum keys to lookupeditor control dynamically in dialog.ts

### DISCUSSION THREAD

**Discussion by rhemasystems** [serenity-is/Serenity#7068] [Help]
Please how can I load enumkey values into a lookupeditor dynamically when another lookupeditor value is changed on the same form.

```
[EnumKey("Smart.Constants")]
    public enum Types
    {
        [Description("Basic")]
        Basic = 1,
        [Description("Secondary")]
        Secondary = 2,
        [Description("College")]
        College = 3,
        [Description("Polytechnic")]
        Polytechnic = 4,
        [Description("University")]
        University = 5
    }

this.form.HeadCountType.changeSelect2(e => {
    
    var headcountid = Q.toId(this.form.HeadCountType.value);
    this.form.HeadCountTypeOption.clearItems();
    if (headcountid != null) {
   //enum key `Types` is loaded into the HeadCountTypeOption control
        this.form.HeadCountTypeOption.addItem(???);
    }
});
```

**Comment by rhemasystems** [comment]
Hi,
Please any form of assistance will be greatly appreciated.
Thank you

**Reply by VictorTomaili** [reply]
https://demo.serenity.is/BasicSamples/SelectWithHardcodedValues

**Comment by rhemasystems** [comment]
Thank you @VictorTomaili , I am familiar with that, what I need help with is how I can load the enum values into the Select2 control

**Reply by VictorTomaili** [reply]
https://github.com/serenity-premium/StartSharp/blob/master/pro-features/src/Serenity.Demo.AdvancedSamples/Modules/Editors/EnumFlagsEditor/EnumFlagsEditorPage.tsx#L27

https://github.com/serenity-is/common-features/blob/master/src/Serenity.Demo.BasicSamples/Modules/Editors/SelectWithHardcodedValues/SelectWithHardcodedValuesPage.ts#L36

**Comment by rhemasystems** [comment]
Thank you @VictorTomaili , but the first link doesn't work, i get **Page Not Found** error.
Thank you

**Reply by VictorTomaili** [reply]
ohh sorry, it is premium feature. I point there `EnumTypeRegistry.get('my.enum.key')` line.
